File: src/main/java/com/example/FleetSystem/model/Vehicle.java
package com.example.FleetSystem.model;

import lombok.*;
import org.hibernate.envers.Audited;
import org.hibernate.envers.NotAudited;

import javax.persistence.*;
import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalDateTime;

import static org.hibernate.envers.RelationTargetAuditMode.NOT_AUDITED;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
@Entity
@Audited
@Table(name = "vehicle")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotAudited
    private String processOrderNumber;
    @NotAudited
    private String plateNumber;
    @NotAudited
    private String make;
    @NotAudited
    private Integer year;
    @NotAudited
    private String design;
    @NotAudited
    private String model;
    @NotAudited
    private String type;
    @NotAudited
    private String capacity;
    @NotAudited
    private String power;
    @NotAudited
    private Date registrationExpiry;
    @NotAudited
    private boolean registrationStatus;
    @NotAudited
    private String fuelType;
    @NotAudited
    private Date insuranceExpiry;
    @NotAudited
    private boolean insuranceStatus;
    @NotAudited
    private Integer leaseCost;
    @NotAudited
    private Date leaseStartDate;
    @NotAudited
    private Date leaseExpiryDate;
    @NotAudited
    private Integer replaceLeaseCost;
    @NotAudited
    private Date replacementDate;
    @NotAudited
    private String usageType;
    @NotAudited
    private String region;
    @NotAudited
    private String location;
    @NotAudited
    private String country;
    @NotAudited
    private String category;
    @NotAudited
    private String costCenter;
    @NotAudited
    private String uuid;
    private String vehicleStatus;
    private String replacementVehicleStatus;
    private String historyStatus;
    private String replacementReason;
    private String replacementRemarks;
    private Integer monthlyRate;

    @NotAudited
    private LocalDateTime createdAt;
    @NotAudited
    private LocalDate updatedAt;

    @OneToOne()
    private Vehicle replacementVehicle;

    @ManyToOne
    @NotAudited
    private Vendor vendor;

    @Audited(targetAuditMode = NOT_AUDITED)
    @ManyToOne
    @JoinColumn(name = "created_by")
    private User createdBy;

    @Audited(targetAuditMode = NOT_AUDITED)
    @ManyToOne
    @JoinColumn(name = "updated_by")
    private User updatedBy;

   }

File: frontend/src/app/modal/vehicle.ts

export interface Vehicle {
    id: Number | null | undefined,
    processOrderNumber: Number | null | undefined
    plateNumber: string | null | undefined;
    make: String | null | undefined
    year: String | null | undefined
    design: String | null | undefined
    model: String | null | undefined
    type: String | null | undefined
    capacity: String | null | undefined
    power: String | null | undefined
    region: String | null | undefined
    country: String | null | undefined
    location: String | null | undefined
    fuelType: String | null | undefined
    costCenter: String | null | undefined
     vendor:
     {
        id: Number| null | undefined
        vendorName: String| null | undefined
        officeLocation: String | null | undefined
        attachments: String | null | undefined

    }
    leaseCost: Number | null | undefined
    replaceLeaseCost: Number | null | undefined
    leaseStartDate: String | Date | null | undefined
    leaseExpiryDate: String | Date | null | undefined
    usageType: String | null | undefined
    category: String | null | undefined
    vehicleStatus: String | null | undefined
    replacementVehicleStatus: String | null | undefined
    registrationStatus: Boolean | null | undefined
    insuranceStatus: Boolean | null | undefined
    replacementDate: Date | null | undefined
    replacementReason: String | null | undefined
    replacementRemarks: String | null | undefined
    replacementVehicle: Vehicle | null | undefined
}
